semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/ICallResolver.scala

```diff
@@ -1,14 +1,45 @@
 package io.shiftleft.semanticcpg.language
 
+import gremlin.scala.GremlinScala
 import io.shiftleft.codepropertygraph.generated.nodes
+import io.shiftleft.codepropertygraph.generated.nodes.{CallRepr, Method}
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
 
 trait ICallResolver {
   def resolveDynamicCallSite(callsite: nodes.Call): Unit
   def resolveDynamicMethodCallSites(method: nodes.Method): Unit
+  def getResolvedCalledMethods(callsite: nodes.CallRepr): Iterable[nodes.Method]
+  def getResolvedCallsites(method: nodes.Method): Iterable[nodes.CallRepr]
+
```

Comment:
We need to decide, document and consolidate the API here:

What triggers a resolve? How can I query side-effect-free the status of the resolution? What status can the resolution be in?

I think resolution can be in status: Not started, started, finished. 

95% of the time, people don't care: They want to "resolve this and don't leak implementation internals", i.e. they don't care about the cache.

As such, I propose to remove `resolveDynamicCallSite` and replace it with `getResolvedCalledMethods` that starts a resolution (returns future/callback or list of results), plus `getResolvedCalledMethodsStatus` that immediately returns the resolution status, without triggering any side-effects.

Reply:
I pushed an adjust version.